packages/wonder-blocks-dropdown/src/components/__tests__/search-text-input.test.js

```diff
@@ -1,145 +1,212 @@
 
-    it("text input should not be focused when losing focus", () => {
+    test("text input should not be focused when losing focus", () => {
         // Arrange
-        const wrapper = shallow(
+        render(
             <SearchTextInput
                 searchText=""
                 testId="search-text-input"
                 onChange={jest.fn()}
             />,
         );
 
-        const input = wrapper.find(`[data-test-id="search-text-input"]`);
+        const input = screen.getByTestId("search-text-input");
         // focus in
-        input.simulate("focus");
+        input.focus();
 
         // Act
         // focus out
-        input.simulate("blur");
+        input.blur();
 
         // Assert
-        expect(wrapper).toHaveState({focused: false});
+        expect(input).not.toHaveFocus();
     });
 
-    it("onChange should be invoked if text input changes", () => {
+    test("onChange should be invoked if text input changes", () => {
         // Arrange
         const onChangeMock = jest.fn();
 
-        const wrapper = shallow(
+        render(
             <SearchTextInput
                 searchText=""
                 testId="search-text-input"
                 onChange={onChangeMock}
             />,
         );
 
-        const input = wrapper.find(`[data-test-id="search-text-input"]`);
+        const input = screen.getByTestId("search-text-input");
 
         // Act
-        input.simulate("change", {
-            target: {value: "query"},
-            preventDefault: jest.fn(),
-        });
-
-        wrapper.update();
+        userEvent.paste(input, "value");
 
         // Assert
-        expect(onChangeMock).toHaveBeenCalledWith("query");
+        expect(onChangeMock).toHaveBeenCalledWith("value");
     });
 
-    it("displays the dismiss button when search text exists", () => {
+    test("displays the dismiss button when search text exists", () => {
         // Arrange
-        const wrapper = shallow(
-            <SearchTextInput
-                searchText="query"
-                testId="search-text-input"
-                onChange={() => jest.fn()}
-                onClick={() => jest.fn()}
-            />,
-        );
+        const SearchFieldWrapper = () => {
+            const [value, setValue] = React.useState("");
+            return (
+                <SearchTextInput
+                    searchText={value}
+                    testId="search-text-input"
+                    onChange={setValue}
+                />
+            );
+        };
+
+        render(<SearchFieldWrapper />);
+
+        const input = screen.getByTestId("search-text-input");
+
+        // Act
+        userEvent.paste(input, "value");
+
+        // Assert
+        const clearIconButton = screen.queryByRole("button");
+        expect(clearIconButton).toBeInTheDocument();
+    });
 
-        const input = wrapper.find(`[data-test-id="search-text-input"]`);
+    test("search should be cleared if the clear icon is clicked", () => {
```

Comment:
You could get rid of this test b/c it's already being tested in `search-field`.

Reply:
I added this test again in this file on purpose, because it initially wasn't working in dropdown but it wasn't showing up as a test failure! If it's okay, I think I'd prefer to keep it.

Reply:
Sounds good :)